app/models/my_module_widget.rb

```diff
@@ -14,4 +14,16 @@ class MyModuleWidget < ActiveRecord::Base
              class_name: 'User',
              inverse_of: :last_modified_my_module_widgets
   belongs_to :my_module, inverse_of: :my_module_widgets
+
+  def destroy
+    delete
+
+    # Update position on other module widgets of this widget's module
```

Comment:
I would put this code into `after_destroy` callback function if it can be done, and remove the `destroy` function (keep its base class definition).